### PR TITLE
Machine learning: classification and regression

### PR DESCRIPTION
**Pull request by anuprulez**
The PR updates the tutorial

### CODE REVIEW

**Comment by bgruening** [comment]
@khanteymoori can you also look at this one?

File: topics/statistics/tutorials/classification_regression/tutorial.md [hunk 1]
```diff
@@ -30,13 +30,21 @@ contributors:
...
+Machine learning is a combined field of computer science, mathematics and statistics to create a predictive model by learning patterns in a dataset. The dataset may have an output field which makes the learning process supervised. The [supervised learning](https://en.wikipedia.org/wiki/Supervised_learning) methods in machine learning have outputs/targets/classes/categories defined in the datasets. These targets can either be discrete (integers) or real (continuous). When the targets are discrete, the learning task is known as classification. Thus, the classification is assigning a class/target to each sample (each row is a sample) in the dataset. The algorithms which are used for this learning task are called classifiers. When the targets are continuous, the learning task is called regression and the algorithms which are used for this task are called regressors. We will go through classification first and look at regression later in this tutorial.
```
Comment (khanteymoori) [inline topics/statistics/tutorials/classification_regression/tutorial.md]: Machine learning is a subset of artificial intelligence (AI) that provides machines the ability to automatically learn from data without being explicitly programmed. Machine learning is a combined field of computer science, mathematics and statistics to create a mathematical model by learning patterns in a dataset.  The [supervised learning](https://en.wikipedia.org/wiki/Supervised_learning) methods is one of the machine learning categories which dataset includes input and desired outputs/targets/classes/categories. These targets can either have limited set of values or any real value. When the targets are limited set of values, the learning task is known as classification. Each row in the dataset is a sample and the classification is assigning a class label/target to each sample. The mathematical model which are used for this learning task are called classifiers. When the targets are real values, the learning task is called regression and the mathematical model which are used for this task are called regressors. We will go through classification first and look at regression later in this tutorial.

File: topics/statistics/tutorials/classification_regression/tutorial.md [hunk 2]
```diff
@@ -47,14 +55,21 @@ Figure [2](#figure-2) shows how a (regression) curve is fit which explains most
...
+[Classification](https://en.wikipedia.org/wiki/Statistical_classification) task assigns a category/class to a sample by learning a decision boundary for samples (rows) in a dataset. This dataset is called a training dataset and contains a class/category for each sample. The algorithm which performs this task is called a classifier. The training dataset contains "features" as columns and a mapping between these features and the target is learned for each sample. The performance of mapping is evaluated using a test dataset (it is separate from training dataset). The test dataset contains only the feature columns and not the target column. The target column is predicted using the mapping learned on the training dataset. In this tutorial, we will use a classifier to train a model using a training dataset, predict the targets for test dataset and visualize the results using plots.
```
Comment (khanteymoori) [inline topics/statistics/tutorials/classification_regression/tutorial.md]: [Classification](https://en.wikipedia.org/wiki/Statistical_classification) task assigns a category/class to each sample by learning a decision boundaries in a dataset. This dataset is called a training dataset and contains samples and desired class/category for each sample. The training dataset contains "features" as columns and a mapping between these features and the target is learned for each sample. The performance of mapping is evaluated using a test dataset (it is separate from training dataset). The test dataset contains only the feature columns and not the target column. The target column is predicted using the mapping learned on the training dataset. In this tutorial, we will use a classifier to train a model using a training dataset, predict the targets for test dataset and visualize the results using plots.

File: topics/statistics/tutorials/classification_regression/tutorial.md [hunk 3]
```diff
@@ -126,89 +155,105 @@ After the training and prediction tasks, we should check whether the predictions
...
+For classification, the targets are discreet. But, when the targets in a dataset are real-values (real numbers), the machine learning task becomes [regression](https://en.wikipedia.org/wiki/Regression_analysis). Each sample in the dataset has a real-valued output or target. Figure [6](#figure-6) shows how a (regression) curve is fit which explains most of the data points (blue balls). Here, the curve is a straight line (red). The regression task is to learn this curve which explains the underlying distribution of the data points. The target for a new sample will lie on the curve learned by the regression task. The algorithms which are used for regression tasks are called regressors. A regressor learns the mapping between the features of a dataset row and its target value. Inherently, it tries to fit a curve for the targets. This curve can be linear (straight line curve) or non-linear. In this part of the tutorial, we will perform regression on [body density](https://rstudio-pubs-static.s3.amazonaws.com/65314_c0d1e5696cdd4e93a3784ea67f9e3d34.html) dataset.
```
Comment (khanteymoori) [inline topics/statistics/tutorials/classification_regression/tutorial.md]: For classification, the targets are limited set of values. But, when the targets in a dataset are real numbers, the machine learning task becomes [regression](https://en.wikipedia.org/wiki/Regression_analysis). Each sample in the dataset has a real-valued output or target. Figure [6](#figure-6) shows how a (regression) curve is fit which explains most of the data points (blue balls). Here, the curve is a straight line (red). The regression task is to learn this curve which explains the underlying distribution of the data points. The target for a new sample will lie on the curve learned by the regression task. A regressor learns the mapping between the features of a dataset row and its target value. Inherently, it tries to fit a curve for the targets. This curve can be linear or non-linear. In this part of the tutorial, we will perform regression on [body density](https://rstudio-pubs-static.s3.amazonaws.com/65314_c0d1e5696cdd4e93a3784ea67f9e3d34.html) dataset.

**Pull request merged**